src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -783,8 +774,8 @@ void CastSpellInfoHelpers::castSpell() {
                     }
 
                     bool has_weak = false;
-                    for (uint pl_id = 0; pl_id < 4; pl_id++) {
-                        if (pParty->pPlayers[pl_id].conditions.Has(Condition_Weak)) {
+                    for (Player &PlayerRef : pParty->pPlayers) {
```

Comment:
Code style: variables start with lowercase, just use `player`. Also you actually need `const Player &` here.